syntax = "proto3";


service Users {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResult);
  rpc GetUsers (GetUsersRequest) returns (stream GetUsersResult);
  rpc Test (TestRequest) returns (stream TestResult);
}
message User {
  string username = 1;
  uint32 user_id = 2;
}

message CreateUserRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message CreateUserResult {
  User user = 1;
}

message GetUsersRequest {
  repeated User user = 1;
}

message GetUsersResult {
  User user = 1;
}
message TestRequest {
  Test test = 1;
}

message TestResult {
  repeated Test test = 1;
}
message Test {
  repeated int32 myint= 1;
}